### ISSUE TITLE
Evaluation on real videos

### ISSUE THREAD

**Issue by sahilg06** [jagnusson/Wav2Lip-Emotion#15]
Hi @jagnusson, @arunasank . Thanks for a nice project.
I tried the inference on a random video using the provided **n2h_full** pretrained model. I used this command
```
python3 real_videos_inference.py --mode dubbed
                                 --results_dir <path to where you want videos to go>
                                 --data_root <path to actual videos to input to generator model>
                                 --checkpoint_path checkpoints/checkpoint_canonical_step000201000.pth
                                 --face_res 96
                                 --min_frame_res 160
```
Using the **affect** mode requires filelist according to your code, so I chose the **dubbed** mode so that it can make the filelist itself from the data_root. (see [here](https://github.com/jagnusson/Wav2Lip-Emotion/blob/244003508c1eb35588c984bfff981b88a2e516f1/evaluation/real_videos_inference.py#L225-L234))
Also, I set the flag for the full mask as True.

But I was really disappointed with the results. Am I doing anything wrong?

This was the input video

https://user-images.githubusercontent.com/59660566/215243343-dce6b797-f13b-4c1b-a704-bfc81a51d115.mp4

And this was the output video

https://user-images.githubusercontent.com/59660566/215243392-324911c9-f95c-4f23-8848-55b9a6df43fb.mp4

Also, there was another problem, I changed the video in the data_root to a video from the **mead** dataset. But this time, it generated the **exactly same frames** that it generated in the case of the above video. What's wrong with the inference code??
Please help!

Input video

https://user-images.githubusercontent.com/59660566/215246562-b2b7869a-45e7-4403-8455-5e65c3fdd34e.mp4

Output video

https://user-images.githubusercontent.com/59660566/215244853-ebb7e58c-bd92-4f27-949b-5e2e19ab8fb0.mp4

**Comment by IanMagnusson** [comment]
Hi @sahilg06,

It looks like you're using the inference code correctly. I think you're seeing some of the issues with generalizability that we also found in our paper. For instance in figure 2 here you can see that some of the models have similarly poor results: 
<img width="1083" alt="Screenshot 2023-01-30 at 9 31 15 AM" src="https://user-images.githubusercontent.com/40903802/215550770-6fc3b3bb-66e5-444e-a259-f788bc0f722e.png">

You might have somewhat more success with a different one of our trained models (perhaps one of the sad -> happy models?) But I wouldn't expect much better performance on these sorts of out of distribution videos. While the second video you have here is from the Mead data, we only trained on the frontal face pose so doing inference on other poses will likely have worse performance. If you are interested in extending this work to make it more generalizable, we think one of the most important limitations of how we trained these models is that they are trained on relatively few distinct faces, so increasing the number of different faces in the training data may improve generalizability.

Thank you for checking out our project!

**Comment by sahilg06** [comment]
Got it. Thank you for your response
